src/Settings/AlgoliaSettingsManager.php

```diff
@@ -49,8 +49,8 @@ public function push(array $params)
 
             if (is_readable($filename)) {
                 $index = $this->algolia->initIndex($indexName);
-                $settings = json_decode(file_get_contents($filename));
-                $index->setSettings($settings);
+                $settings = json_decode(file_get_contents($filename), true);
+                $index->setSettings($settings)->wait();
```

Comment:
This could be pretty long, especially for large index. I don't know if we should wait here, or maybe it could be an option but not wait by default. 🤷‍♂ Let's see what others think.

Reply:
I see, I think for tests, we could use the SyncClient to make sure every call followed by a `wait()`.

Reply:
The SyncClient is not present in v2 anymore. I'll change the tests accordingly